### PR TITLE
Add workflows and exchanges endpoints.

### PR DESCRIPTION
**Pull request by wes-smith**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/pull/382.html" title="Last updated on May 29, 2024, 8:35 PM UTC (dee75b2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/382/5099989...dee75b2.html" title="Last updated on May 29, 2024, 8:35 PM UTC (dee75b2)">Diff</a>

### CODE REVIEW

File: exchanges.yml [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+              description: The step from the above set that the exchange starts on.
+        options:
+          type: object
+          properties:
+            controller:
+              type: string
+              description: The controller of the instance.
+            authorization:
+              type: object
+              description: Authorization scheme information (e.g. OAuth2 configuration).
```
Comment (TallTed) [inline exchanges.yml:231]:
```suggestion
              description: Authorization scheme information (e.g., OAuth2 configuration).
```

File: exchanges.yml [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
```
Comment (TallTed) [inline exchanges.yml:179]:
```suggestion
              description: Authorization scheme information (e.g., OAuth2 configuration).
```

**Review by TallTed — CHANGES_REQUESTED** [review]
trivial, human-facing

File: exchanges.yml [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+            controller:
+              type: string
+              description: The controller of the instance.
+            authorization:
+              type: object
+              description: Authorization scheme information (e.g., OAuth2 configuration).
+            credentialTemplates:
+              type: array
+              description: One or more VC templates for issuance.
+              items:
```
Comment (brianorwhatever) [inline exchanges.yml:245]: This should reference to the `VerifiableCredentialTemplate` which will fix one of the issues in the failing github action

File: exchanges.yml [hunk 4]
```diff
@@ -0,0 +1,396 @@
+openapi: 3.0.0
+info:
+  version: "0.0.3-unstable"
+  title: VC Exchanges API
```
Comment (brianorwhatever) [inline exchanges.yml:4]: This title misses the "workflow" side of things

File: exchanges.yml [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CreateWorkflowResponse"
+        "400":
+          description: Invalid input
+        "401":
+          description: Not Authorized
+        "500":
+          description: Internal Error
+  /workflows/{localWorkflowId}:
```
Comment (brianorwhatever) [inline exchanges.yml:42]: `localWorkflowId` must be defined in a `parameters` section below this

File: exchanges.yml [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetWorkflowResponse"
+        "400":
+          description: Invalid input
+        "401":
+          description: Not Authorized
+        "500":
+          description: Internal Error
+  /workflows/{localWorkflowId}/exchanges:
```
Comment (brianorwhatever) [inline exchanges.yml:66]: `localWorkflowId` must be defined in a `parameters` section below this

File: exchanges.yml [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/CreateExchangeResponse"
+        "400":
+          description: Invalid input
+        "401":
+          description: Not Authorized
+        "500":
+          description: Internal Error
+  /workflows/{localWorkflowId}/exchanges/{localExchangeId}:
```
Comment (brianorwhatever) [inline exchanges.yml:95]: `localWorkflowId` and `localExchangeId` must be defined in a `parameters` section below this

**Review by brianorwhatever — CHANGES_REQUESTED** [review]
The OpenAPI linting is currently failing and I have pointed out some fixes required as well as the following:
- The **no-empty-servers** rule is triggered by this however I don't think we should include a server in this repo. In fact we should remove the servers from the other API files
- The security schemes seem fine to me however they are triggering the **security-defined** rule. I am not sure why it is allowing them in other files (maybe it just isn't checking them as the files aren't changing) however they look the same here as they do in the other API files.

File: exchanges.yml (hunk 3 above)
Comment (wes-smith) [reply exchanges.yml:245]: Removed the template. Rendering is having an issue here, but there appears to be a ReSpec issue hurting rendering in general, so unclear if this is incorrect formatting or a consequence of a separate issue.

Comment (wes-smith) [reply exchanges.yml:252]: `no-empty-servers` and `security-defined` rules still causing linting errors. Not sure how to proceed.

Comment (brianorwhatever) [reply exchanges.yml:252]: I believe the ` security-defined` would be solved by adding [this](https://github.com/w3c-ccg/vc-api/blob/5099989349afa55237ee70c5d5f18c8d90664e13/issuer.yml#L74C1-L75C73)

and maybe we can just add a fake url to the servers section?

Comment (wes-smith) [reply exchanges.yml:257]: Listing errors fixed, placeholder URL added. Thanks!

**Review by brianorwhatever — APPROVED** [review]

**Pull request merged**